dds/DCPS/transport/framework/TransportRegistry.cpp

```diff
@@ -157,7 +157,8 @@ TransportRegistry::load_transport_configuration(const OPENDDS_STRING& file_name,
             // Create the TransportInst object and load the transport
             // configuration in ACE_Configuration_Heap to the TransportInst
             // object.
-            TransportInst_rch inst = create_inst(ACE_TEXT_ALWAYS_CHAR(transport_id.c_str()), transport_type);
+            const OPENDDS_STRING tid_str = ACE_TEXT_ALWAYS_CHAR(transport_id.c_str());
```

Comment:
I think Coverity will still complain about this, since the result of c_str() is still directly passed into the constructor and/or operator=() of basic_string, either of which will dereference it. I think it just wants to see you checking for null before using it?

Perhaps something like:
`const OPENDDS_STRING tid_str = transport_id.c_str() ? ACE_TEXT_ALWAYS_CHAR(transport_id.c_str()) : OPENDDS_STRING();`

Reply:
Or an if statement after declaring it, if that feels cleaner to you. I realize having OPENDDS_STRING() in there looks hokey.

Reply:
How about `const OPENDDS_STRING tid_str = transport_id.c_str() ? ACE_TEXT_ALWAYS_CHAR(transport_id.c_str()) : "";`?